System/Primitive.cpp

```diff
@@ -288,6 +288,40 @@ namespace RTE {
 		}
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void EllipsePrimitive::Draw(BITMAP *drawScreen, Vector targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			ellipse(drawScreen, drawStart.m_X, drawStart.m_Y, m_HorizRadius, m_VertRadius, m_Color);
+		} else {
+			Vector drawStartLeft;
+			Vector drawStartRight;
+
+			TranslateCoordinates(targetPos, m_StartPos, drawStartLeft, drawStartRight);
+
+			ellipse(drawScreen, drawStartLeft.m_X, drawStartLeft.m_Y, m_HorizRadius, m_VertRadius, m_Color);
```

Comment:
This works properly? Drawing 2 ellipses and just shoving them together like this? I guess anything off the edge of the scene isn't visible so it doesn't matter??

Reply:
This is literally how everything else is handled so it doesn't shit itself when passing the seam.